pkg/strutil/merge.go

```diff
@@ -12,7 +12,7 @@ import (
 // while removing all duplicates.
 func MergeSlices(a ...[]string) []string {
 	if len(a) == 0 {
-		return nil
+		return make([]string, 0)
```

Comment:
This really not the best pattern, can we avoid this?

Reply:
Okay, so I did this because when there isn't a store API connected the Thanos Querier, it returns `{ "data": null }` because the function `MergeSlices` returns `nil`. This causes the UI to error out.

What I wanted to do was to return `{ "data": [] }` instead, so that we don't get an error in the UI saying `data` is null, expected array, and as we can see in this playground https://play.golang.org/p/zDLyXcc8ie the only way to return an empty array is to use `make` to create the array.

Can you suggest me some alternative to this? One solution that came to my mind is to check if the array is nil and then return an empty array instead in the `pkg/query/api/v1.go`
```go

func (api *API) labelNames(r *http.Request) (interface{}, []error, *ApiError) {
	// redacted

	names, warnings, err := q.LabelNames()
	if err != nil {
		return nil, nil, &ApiError{errorExec, err}
	}

	// names would be nil in case there is no store api connected to querier
	if names == nil {
		names = make([]string, 0)
	}

	return names, warnings, nil
}
```

Reply:
Yes, it would be nice to have some layer of conversion on UI side. The reason is that we will not remember to put everywhere allocated slice as it's not something you do in Go, because you just kind of waste CPU / mem. (: So would be nice to handle that on UI layer globally. WDYT?

Reply:
Okay, got it. I'll add a null check on the UI side. But as we are using almost the same code for the graph page as upstream, I think it's not a great idea to significantly diverge from the code. Maybe we can upstream this null check too. Also I guess it's fine to go with
```go
if names == nil {
	names = make([]string, 0)
}
```
in `api.go`, this way we won't have an allocated slice for any other consumer of `MergeSlice` function, but only in `labelNames` and that too only when we need it.

Reply:
@bwplotka the convention is to return empty arrays in such cases: https://stackoverflow.com/questions/6045895/should-a-restful-api-return-404-for-arrays-of-objects, https://stackoverflow.com/questions/13366730/proper-rest-response-for-empty-table, https://groups.google.com/forum/#!topic/api-craft/wngl_ZKONyk, etc.

Reply:
Ah, I see, this was added to our utility function, that's why you had commented. My bad.